vms/platformvm/vm_test.go

```diff
@@ -304,8 +314,43 @@ func BuildGenesisTestWithArgs(t *testing.T, args *api.BuildGenesisArgs) (*api.Bu
 	return &buildGenesisArgs, genesisBytes
 }
 
-func defaultVM(t *testing.T) (*VM, database.Database, *mutableSharedMemory) {
+func defaultVM(t *testing.T, fork activeFork) (*VM, database.Database, *mutableSharedMemory) {
 	require := require.New(t)
+	var (
+		apricotPhase3Time = mockable.MaxTime
+		apricotPhase5Time = mockable.MaxTime
+		banffTime         = mockable.MaxTime
+		cortinaTime       = mockable.MaxTime
+		dTime             = mockable.MaxTime
+	)
+
+	// always reset latestForkTime (a package level variable)
+	// to ensure test independency
+	latestForkTime = defaultGenesisTime.Add(time.Second)
+	switch fork {
+	case apricotPhase3:
+		apricotPhase3Time = latestForkTime
```

Comment:
So all forks are inactive by default, because they are initialized at `mockable.MaxTime`. Depending on which one is activated by this function client, we reset that fork and all preceding ones to a finite time. For simplificy we activate all of them at once, so they have the same activation time

Reply:
can't we define an order between upgrades (like an ordered list) and activate previous upgrades? It could at least make it easier to define a new upgrade.

I liked the idea of specifying upgrades as param.

Reply:
Upgrades as params are mandatory for maintanability I believe. There are other PRs where I explicitly set tests over the latestFork so to be able to upgrade them easily once a new fork come.
I'd move the ordered upgrades list to the [thread the cleans up the UTs](https://github.com/ava-labs/avalanchego/issues/2453) if it's ok with you.
This PR is an extract from that thread with the bare minimum I need for Durango tests.